### PR TITLE
Amend `KConfig` menu and disable developer fun

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

**Review by danielskinstad — APPROVED** [review]
Looks good to me

File: prj.conf [hunk 1]
```diff
@@ -92,13 +92,13 @@ CONFIG_DNS_NUM_CONCUR_QUERIES=5
...
+# CONFIG_DEBUG_COREDUMP_BACKEND_LOGGING=y
+# CONFIG_DEBUG_COREDUMP_MEMORY_DUMP_LINKER_RAM=n
+# CONFIG_DEBUG_COREDUMP_MEMORY_DUMP_MIN=y
 # Asserts
 # https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_ASSERT_VERBOSE
-CONFIG_ASSERT=y
-CONFIG_ASSERT_VERBOSE=y
-CONFIG_PICOLIBC_USE_MODULE=y
-CONFIG_PICOLIBC_ASSERT_VERBOSE=y
+# CONFIG_ASSERT=y
```
Comment (vpodzime) [inline prj.conf:101]: Shouldn't we keep this one?

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise.

File: prj.conf (hunk 1 above)
Comment (lluiscampos) [reply prj.conf:101]: I don't have an strong opinion. I added all all these at the beginning but got little value out of it.

I'll leave the `CONFIG_ASSERT` based in your comment.

Comment (lluiscampos) [reply prj.conf]: Adding _only_ this configuration option [broke the build](https://gitlab.com/Northern.tech/Mender/mender-mcu-integration/-/jobs/8248493185) :laughing:

Comment (lluiscampos) [reply prj.conf]: Pushed a workaround for this.

**Pull request merged**